Starting Files for the Program

Icefireburn1 opened the pull request:
Adding some essential files for the program.

Icefireburn1 committed:
Adding the starting files for the project.

Icefireburn1 committed:
Fixed a spelling error

JimLeaf commented on src/App.tsx:
```diff
@@ -0,0 +1,78 @@
+    const [currentTab, setCurrentTab] = useState(0);
+    const [searchText, setSearchText] = useState("");
+
+    function handleStatusChange(status: boolean) {
```
dead code? not called

Icefireburn1 replied:
Removed the empty code.

JimLeaf commented on src/App.tsx:
```diff
@@ -0,0 +1,78 @@
+        </div>
+      );
+}
+/*
```
Don't check in commented out code / dead code

Icefireburn1 replied:
Remove useful commented code.

JimLeaf commented on src/App.tsx:
```diff
@@ -0,0 +1,78 @@
+      setCurrentTab(newValue);
+    };
+
+    const something=(event: any)=> {
```
name event handler function better

Icefireburn1 replied:
Renamed it to be more relevant.

JimLeaf commented on src/App.tsx:
```diff
@@ -0,0 +1,78 @@
+    }
+      return (
+        <div className="App" 
+            onKeyDown={(e) => {
```
Probably unnecessary delegate function wrapper, try updating to just assigning onKeyDown to your handler function

Icefireburn1 replied:
Did as suggested.

JimLeaf commented on src/components/EssayForm.js:
```diff
@@ -0,0 +1,51 @@
+      super(props);
+      this.state = {
+        initialValue: this.props.value,
+        value: '\nEric is gay',
```
Set better default value

Icefireburn1 replied:
Changed the default value.

JimLeaf commented on src/components/EssayForm.js:
```diff
@@ -0,0 +1,51 @@
+class EssayForm extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
```
Decide on a consistent style across all of the code.  I.e. either use "this." where applicable in all code or nowhere.  But don't use it sometimes and not in other files.

JimLeaf commented on src/components/PokemonDisplay.js:
```diff
@@ -0,0 +1,107 @@
+
+    async GetWebsiteHTML(webUrl)
+     {
+        var result = await $.ajax({ 
```
Worth investigating if the electronjs framework has classes for web requests, and consider using those instead of ajax.  Likewise when going to potentially save local copy of dictionary to a file look to use electron file system classes.

JimLeaf commented on src/components/PokemonDisplay.js:
```diff
@@ -0,0 +1,107 @@
+            index: 0,
+            value: "",
+            pokemon: {
+                name:				"",
```
No need for formatted tabbing / spacing.  Just single space

JimLeaf commented on src/components/PokemonDisplay.js:
```diff
@@ -0,0 +1,107 @@
+         var tableStr = tableHtml.find('.vitals-table');
+    
+        var loadedPokemon = {
+            name:				tableHtml.find('title')[0].innerHTML.replace(/ .*/,''),
```
No need for formatted tabbing / spacing.  Just single space

JimLeaf commented on src/components/PokemonDisplay.js:
```diff
@@ -0,0 +1,107 @@
+        this.state = {
+            index: 0,
+            value: "",
+            pokemon: {
```
Consider making a Pokemon interface / class. And move its properties and the html parsing logic to it

Icefireburn1 replied:
Moved all Pokemon related information/methods to their own class.

JimLeaf commented on src/components/PokemonDisplay.js:
```diff
@@ -0,0 +1,107 @@
+         var tableHtml = $('<table>', {html:pageSource});
+         var tableStr = tableHtml.find('.vitals-table');
+    
+        var loadedPokemon = {
```
Probably want to make a pokemon data parsing method and have it handle negative cases and returning appropriate errors if the pageSource html changes in the future

Icefireburn1 replied:
For the moment, added some simple error handling.

JimLeaf commented on src/components/SearchBar.tsx:
```diff
@@ -0,0 +1,86 @@
+            autoSelect
+            classes={{inputRoot: classes.inputRoot, root: classes.root}}
+            onInputChange={(e,v,s) => input = v}
+            options={pokemonList}
```
TODO: pull pokemonList from an async http request?

Icefireburn1 replied:
Finished.

JimLeaf commented on src/components/Clickable.js:
```diff
@@ -0,0 +1,23 @@
+import React, {Component} from 'react'
```
Change all the '.js' file extensions to '.jsx' for consistency

JimLeaf commented on src/components/Clickable.js:
```diff
@@ -0,0 +1,23 @@
+import React, {Component} from 'react'
+import Button from '@material-ui/core/Button'
```
Include all the other files in your src, public, and root directories. excluding the node_modules folder.  So that anyone who checks out the repository will be able to build/launch

Icefireburn1 replied:
I think I got all of the needed files included now!

JimLeaf reviewed (changes requested):
See comments.

Icefireburn1 committed:
Renamed some .js files into .jsx files.
Added a lot of files needed to file the program.
Modified PokemonDisplay to have its own Pokemon class.
Removed unused code.

JimLeaf commented on package.json:
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "cra-craco-electron-example",
```
Give it a real name

JimLeaf commented on package.json:
```diff
@@ -0,0 +1,61 @@
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+ssh://git@github.com:wwlib/cra-craco-electron-example.git"
```
Update to your repo or just remove the entire repository block here

JimLeaf commented on package.json:
```diff
@@ -0,0 +1,61 @@
+  "keywords": [
+    "react,create-react-app,cra,craco,electron,electron-builder"
+  ],
+  "author": "Andrew Rapo <andrew@wwlib.org> (http://wwlib.org/)",
```
update or remove

JimLeaf commented on public/index.html:
```diff
@@ -0,0 +1,40 @@
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Update to what you want window to show

JimLeaf commented on public/manifest.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
```
update

JimLeaf commented on src/App.tsx:
```diff
@@ -0,0 +1,64 @@
+      setCurrentTab(newValue);
+    };
+
+    const EnterKeyResponse=(event: any)=> {
```
nit pick: do consistent camel casing for method names (i.e. enterKeyResponse)

JimLeaf commented on src/components/PokemonDisplay.jsx:
```diff
@@ -0,0 +1,128 @@
+import React, {setState, useEffect} from 'react';
+import $ from 'jquery'
+
+class Pokemon {
```
move Pokemon to its own Pokemon.jsx file.  this will make it so you can include it anywhere separately

JimLeaf commented on src/components/PokemonDisplay.jsx:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    // Try to check for unuseful results due to webpage changes
+    CheckForFaultyResults()
```
nit: camel case

JimLeaf commented on src/components/PokemonDisplay.jsx:
```diff
@@ -0,0 +1,128 @@
+            console.log("Error: National Pokedex Number is NaN");
+    }
+
+    ParseWebpage(pageSource)
```
nit: camel case

JimLeaf reviewed (changes requested):
Last couple of style and comments. In general just try and make sure all the classes look at feel like the same style of code (casing, spacing, how state is set, etc)

Icefireburn1 committed:
Revamped UI.
Added automatic pull for auto suggestions for pokemon.

Icefireburn1 committed:
Removed commented code

Icefireburn1 committed:
Removed unneeded code
Moved Pokemon class into separate file
Moved to camel casing naming convention
